### DISCUSSION TITLE
How do RSF account for censored data when estimating the CHF?

### DISCUSSION THREAD

**Discussion by JuicyCodeHere** [kogalur/randomForestSRC#379] [General]
Dear All,

I have a conceptual question: how do RSF account for censored data when estimating the CHF?
I understand that censoring is accounted for by using a splitting rule that does so, such as the logrank test. 
However, when estimating the CHF of a case we simply go to its terminal node h and divide the number of deaths by the number of individuals at risk (also not sure what that means: all people in h that have not died yet?):

![Screenshot 2023-05-09 at 14 26 35](https://github.com/kogalur/randomForestSRC/assets/88374713/2add78ab-4950-4032-ac87-c0edc0620394)

If that is the case, the RSF does not differentiate between censored and uncensored cases when estimating the CHF right? Isn't that problematic? I am probably misunderstanding something. Thanks already!

**Comment by ishwaran** [comment]
The TN (terminal node estimators) are the Kaplan-Meier (KM) survival function and Nelson-Aalen (NA) cumulative hazard function estimators.  These estimators explicitly account for censoring.  

In your question above you have described the NA estimator.  The number at risk are those who are still alive up to a given time (so this could be cases that get censored but are not yet censored).  In this way censoring is accounted for.   The NA estimator is the nonparametric estimator in right-censored survival settings.

It seems you are unfamiliar with survival analysis therefore I recommend reading some online material related to KM and NA - this might help for you to understand the concepts involved and why the NA estimator you wrote down makes sense as an estimator that accounts for censoring.

**Reply by JuicyCodeHere** [reply]
Hey Dr. Ishwaran, thanks for the swift reply! Indeed I am still wrapping my mind around the concepts, thanks for being patient. I will look into these estimators more but what you are saying makes sense. So if a case is already censored at time t, the person is considered "dead" rather than alive/ at risk? Then censored cases and "dead" cases are treated the same or not? I can look into it myself but thanks already.

**Comment by ishwaran** [comment]
If a case is censored before time t, then they do not contribute to the risk set (the denominator) at time t. Also it is important to recognize that only deaths contribute to the numerator (at no time is a censored case considered as a death).   This is why the KM and NA estimators take jumps at the event times (i.e the observed death times for those patients that die).  The role of the censored cases is to adjust the risk set (those still at risk) in order to get a correct estimator for the hazard, the instantaneous risk of death, which is d_{l,h}/Y_{l,h}, the numerator divided by the denominator.

**Reply by JuicyCodeHere** [reply]
Hey Dr. Ishwaran, I think thats clear now, thank you. It wasn't entirely clear to me which cases are part of the risk set but now I know. Thank you.
